miden-lib/src/transaction/mod.rs

```diff
@@ -174,35 +175,40 @@ impl TransactionKernel {
     ///
     /// The data on the stack is expected to be arranged as follows:
     ///
-    /// Stack: [CNC, FAH]
+    /// Stack: [CNC, FAH, block_expiration_height]
     ///
     /// Where:
     /// - CNC is the commitment to the notes created by the transaction.
     /// - FAH is the final account hash of the account that the transaction is being executed
     ///   against.
+    /// - block_expiration_height is the block height at which the transaction will become expired,
+    ///   defined by the sum of the execution block ref and the transaction's block expiration delta
+    ///   (if set during transaction execution).
     ///
     /// # Errors
     /// Returns an error if:
     /// - Words 3 and 4 on the stack are not 0.
     /// - Overflow addresses are not empty.
     pub fn parse_output_stack(
         stack: &StackOutputs,
-    ) -> Result<(Digest, Digest), TransactionOutputError> {
+    ) -> Result<(Digest, Digest, Option<u32>), TransactionOutputError> {
         let output_notes_hash = stack
             .get_stack_word(OUTPUT_NOTES_COMMITMENT_WORD_IDX * 4)
             .expect("first word missing")
             .into();
+
         let final_account_hash = stack
             .get_stack_word(FINAL_ACCOUNT_HASH_WORD_IDX * 4)
             .expect("second word missing")
             .into();
 
-        // make sure that the stack has been properly cleaned
-        if stack.get_stack_word(8).expect("third word missing") != EMPTY_WORD {
-            return Err(TransactionOutputError::OutputStackInvalid(
-                "Third word on output stack should consist only of ZEROs".into(),
-            ));
-        }
+        let expiration_block_num = stack
+            .get_stack_item(EXPIRATION_BLOCK_ELEMENT_IDX)
+            .expect("element on index 8 missing");
+
+        let expiration_block_num =
+            (expiration_block_num != Felt::ZERO).then_some(expiration_block_num.as_int() as u32);
```

Comment:
There could be some extra validation here to make sure `expiration_block_num` is within 255 units from the block ref, but then it would have to be an input of this (and upstream) fuctions. Maybe it would just be simpler to output the delta?

Reply:
I think the way you have it is fine. Outputting just the delta would make thing a bit less convenient - e.g., to figure out the absolute height, we'd need to get the block height associated with `ProvenTransaction::block_ref` and this may not always be straight-forward.